coretex/cli/modules/utils.py

```diff
@@ -17,20 +17,79 @@
 
 from typing import List, Any, Optional, Callable
 from functools import wraps
+from importlib.metadata import version as getLibraryVersion
 
 import sys
+import logging
 
 from py3nvml import py3nvml
 
 import click
+import requests
 
+from . import ui
 from ...utils.process import command
 
 
 def updateLib() -> None:
     command([sys.executable, "-m", "pip", "install", "--no-cache-dir", "--upgrade", "coretex"], ignoreStdout = True, ignoreStderr = True)
 
 
+def fetchLatestVersion() -> Optional[str]:
+    url = "https://pypi.org/pypi/coretex/json"
+    response = requests.get(url)
+
+    if not response.ok:
+        logging.getLogger("cli").debug(f"Failed to fetch version of coretex library. Response code: {response.status_code}")
+        return None
+
+    data = response.json()
+    infoDict = data.get("info")
+    if not isinstance(infoDict, dict):
+        logging.getLogger("cli").debug("Value of json field of key \"info\" in \"data\" dictionary is not of expected type (dict).")
+        return None
+
+    version = infoDict.get("version")
+    if not isinstance(version, str):
+        logging.getLogger("cli").debug("Value of json field of key \"version\" in \"info\" dictionary is not of expected type (str).")
+        return None
+
+    return version
+
+
+def isVersionStrValid(version: str) -> bool:
+    parts = version.split('.')
+
+    if len(parts) != 3:
+        return False
+
+    return all(part.isdigit() for part in parts)
+
+
+def checkLibVersion() -> None:
+    current = getLibraryVersion("coretex")
```

Comment:
Create a separate function to get the current version which returns a tuple as well